source/code/plugins/in_dsc_checks.rb

```diff
@@ -0,0 +1,114 @@
+require 'fluent/input'
+require 'fluent/config/error'
+
+module Fluent
+  class DscInput < Input
+    Fluent::Plugin.register_input('dsc_checks', self)
+
+    config_param :tag, :string, :default=>nil
+    config_param :check_install_interval, :time, :default=>86400
+    config_param :check_status_interval, :time, :default=>1800
+   
+    CHECK_IF_DPKG = (%x(which dpkg > /dev/null 2>&1; echo $?)).to_i
+
+    def configure(conf)
+      super
+      if !@tag 
+        raise Fluent::ConfigError, "'tag' option is required on dsc_checks input"
+      end
+    end
+
+    def start
+      super
+      @finished_check_install = false
+      @finished_check_status = false
+      @thread_check_install = Thread.new(&method(:run_check_install))
+      @thread_check_status = Thread.new(&method(:run_check_status))
+      @dsc_cache = DSCcache.new
+    end
+
+    def check_install
+      if CHECK_IF_DPKG == 0
+        %x(dpkg --list omsconfig > /dev/null 2>&1; echo $?).to_i
+      else
+        %x(rpm -qi omsconfig > /dev/null 2>&1; echo $?).to_i
+      end
+    end
+
+    def run_check_install
+      until @finished_check_install
+        install_status = check_install
+        @dsc_cache.set("install", install_status)
+        sleep @check_install_interval
+      end
+    end
+
+    def get_dsc_status
+      begin
+        dsc_status = %x(/opt/microsoft/omsconfig/Scripts/PerformRequiredConfigurationChecks.py)
+      rescue => error
+        OMS::Log.error_once("Unable to run PerformRequiredConfigurationChecks.py for dsc : #{error}")
+      end
+      if dsc_status.include?("ReturnValue=0")
```

Comment:
This is a sample error state:
instance of TestConfiguration
{
    ReturnValue=0
    InDesiredState=false
    ResourceId={}
}